spec/routemaster/integration/api_client_spec.rb

```diff
@@ -297,14 +298,15 @@ def timestamp ; subject.get(url).body.updated_at ; end
 
     it 'traverses through pagination next all links that match the request params' do
       res = subject.discover(url)
+
       expect(res.resources.index(filters: { first_name: 'roo' }).count).to eq(5)
     end
 
-    it 'does not make any http requests to fetch resources any if just the index method is called' do
-      resources = subject.discover(url).resources
+    it 'does not make any http requests to fetch individual resources if just the index method is called' do
```

Comment:
As far as I can see the actual caching of the root endpoint isn't tested. Given this caused issues with clients I think we need a test to avoid regressions.